setup.py

```diff
@@ -162,13 +180,14 @@ def run(self):
                         for dependency in cuda_dependencies:
                             if dependency in line:
                                 if dependency not in to_preload:
-                                    to_preload.append(line)
+                                    to_preload_cuda.append(line)
                                 args.extend(['--remove-needed', line])
                     args.append(dest)
                     if len(args) > 3:
                         subprocess.run(args, check=True, stdout=subprocess.PIPE)
 
                 self._rewrite_ld_preload(to_preload)
+                self._rewrite_ld_preload_cuda(to_preload_cuda)
```

Comment:
is it ok to call rewrite_ld_preload() and rewrite_ld_preload_cuda() back to back?
can there be multiple LD_PRELOAD_BEGIN_MARK in the rewritten file and does that work?

Reply:
Ah you're right, the second time through it will erase the original _rewrite_ld_preload section since it's not appending, it just copies until that marker then only writes the new stuff.

Is there a better way to do this? We really just want to append to the end of this file, my python experience is very limited.

Reply:
if it's overwriting, then we lose the stuff from rewrite_ld_preload(to_preload) ? 
if you want to append to end of file, then open with open('file', 'a') instead of open('file', 'w')

Reply:
https://stackoverflow.com/a/30566011 is helpful

Reply:
Yeah, that is much simpler if it works properly. I'm not sure why the original code read in the original file into a buffer then read line by line until it hit the marker vs just appending.